test/azure/terraform_azure_network_example_test.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformAzureNetworkExample(t *testing.T) {
+	t.Parallel()
+
+	// Create values for Terraform
+	subscriptionID := "" // subscriptionID is overridden by the environment variable "ARM_SUBSCRIPTION_ID"
+	uniquePostfix := random.UniqueId()
+	expectedLocation := "eastus2"
+	expectedSubnetRange := "10.0.20.0/24"
+	expectedPrivateIP := "10.0.20.5"
+	expectedDnsIp01 := "10.0.0.5"
+	expectedDnsIp02 := "10.0.0.6"
+	exectedDNSLabel := fmt.Sprintf("dns-terratest-%s", strings.ToLower(uniquePostfix)) // only lowercase, numeric and hyphens chars allowed for DNS
+
+	// Configure Terraform setting up a path to Terraform code.
+	terraformOptions := &terraform.Options{
+		// Relative path to the Terraform dir
+		TerraformDir: "../../examples/azure/terraform-azure-network-example",
+
+		// Variables to pass to our Terraform code using -var options
+		Vars: map[string]interface{}{
+			"postfix":           uniquePostfix,
+			"subnet_prefix":     expectedSubnetRange,
+			"private_ip":        expectedPrivateIP,
+			"dns_ip_01":         expectedDnsIp01,
+			"dns_ip_02":         expectedDnsIp02,
+			"location":          expectedLocation,
+			"domain_name_label": exectedDNSLabel,
+		},
+	}
+
+	// At the end of the test, run `terraform destroy` to clean up any resources that were created
+	defer terraform.Destroy(t, terraformOptions)
+
+	// Run `terraform init` and `terraform apply`. Fail the test if there are any errors
+	terraform.InitAndApply(t, terraformOptions)
+
+	// Run `terraform output` to get the values of output variables
+	expectedRgName := terraform.Output(t, terraformOptions, "resource_group_name")
+	expectedVNetName := terraform.Output(t, terraformOptions, "virtual_network_name")
+	expectedSubnetName := terraform.Output(t, terraformOptions, "subnet_name")
+	expectedPublicAddressName := terraform.Output(t, terraformOptions, "public_address_name")
+	expectedPrivateNicName := terraform.Output(t, terraformOptions, "network_interface_internal")
+	expectedPublicNicName := terraform.Output(t, terraformOptions, "network_interface_external")
+
+	// Tests are separated into subtests to differentiate integrated tests and pure resource tests
+
+	// Integrated network resource tests
+	t.Run("VirtualNetwork_Subnet", func(t *testing.T) {
+		// Check the Subnet exists in the Virtual Network Subnets with the expected Address Prefix
+		actualVnetSubnets := azure.GetVirtualNetworkSubnets(t, expectedVNetName, expectedRgName, subscriptionID)
+		assert.NotNil(t, actualVnetSubnets[expectedSubnetName])
+		assert.Equal(t, expectedSubnetRange, actualVnetSubnets[expectedSubnetName])
+	})
+
+	t.Run("NIC_PublicAddress", func(t *testing.T) {
+		// Check the internal network interface does NOT have a public IP
+		actualPrivateIPOnly := azure.GetNetworkInterfacePublicIPs(t, expectedPrivateNicName, expectedRgName, subscriptionID)
+		assert.True(t, len(actualPrivateIPOnly) == 0)
```

Comment:
Use `assert.Equal`. That way, when there is an error, you get to see the actual value in the error message.
```suggestion
		assert.Equal(t, 0, len(actualPrivateIPOnly))
```

Reply:
Fixed.